Services/CO.CDP.Forms.WebApi/UseCase/UpdateFormSectionAnswersUseCase.cs

```diff
@@ -23,32 +23,19 @@ public async Task<bool> Execute((Guid formId, Guid sectionId, Guid answerSetId,
         var section = await formRepository.GetSectionAsync(formId, sectionId)
             ?? throw new UnknownSectionException($"Unknown section {sectionId} in form {formId}.");
 
+        var sharedConsent = await formRepository.GetSharedConsentDraftWithAnswersAsync(formId, organisationId)
+                            ?? CreateSharedConsent(organisation, section.Form);
 
         var questionDictionary = section.Questions.ToDictionary(q => q.Guid);
 
         ValidateQuestions(answers, questionDictionary);
 
-        var existingAnswerSet = await formRepository.GetFormAnswerSetAsync(sectionId, organisationId, answerSetId);
+        var answerSet = sharedConsent.AnswerSets.FirstOrDefault(a => a.Guid == answerSetId)
+            ?? CreateAnswerSet(answerSetId, sharedConsent, section);
+        answerSet.Answers = MapAnswers(answers, questionDictionary, answerSet.Answers);
 
-        await UploadFileIfRequired(answers, questionDictionary, existingAnswerSet);
-
-        if (existingAnswerSet != null)
-        {
-            existingAnswerSet.Answers = MapAnswers(answers, questionDictionary, existingAnswerSet.Answers);
-        }
-        else
-        {
-            existingAnswerSet = new Persistence.FormAnswerSet
-            {
-                Guid = answerSetId,
-                OrganisationId = organisation.Id,
-                Organisation = organisation,
-                Section = section,
-                Answers = MapAnswers(answers, questionDictionary, []),
-            };
-        }
-
-        await formRepository.SaveAnswerSet(existingAnswerSet);
+        await UploadFileIfRequired(answers, questionDictionary, answerSet);
```

Comment:
UploadFileIfRequired will not work now, it should be done before Answers are mapped.

Reply:
Updated, thanks!